### PR TITLE
[CELEBRON-1282] Optimize push data replica error message

### PR DESCRIPTION
**Pull request by lyy-pineapple**
### What changes were proposed in this pull request?
Optimize the handling of exceptions during the push of replica data, now only throwing PUSH_DATA_CONNECTION_EXCEPTION_REPLICA in specific scenarios.

### Why are the changes needed?
When handling exceptions related to pushing replica data in the worker, unmatched exceptions, such as 'file already closed,' are uniformly transformed into REPLICATE_DATA_CONNECTION_EXCEPTION_COUNT and returned to the client. The client then excludes the peer node based on this count, which may not be appropriate in certain scenarios. For instance, in the case of an exception like 'file already closed,' it typically occurs during multiple splits and commitFile operations. Excluding a large number of nodes under such circumstances is clearly not in line with expectations.
image

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
through exist uts

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala [hunk 1]
```diff
@@ -144,6 +145,7 @@ object WorkerSource {
   val REPLICATE_DATA_WRITE_FAIL_COUNT = "ReplicateDataWriteFailCount"
   val REPLICATE_DATA_CREATE_CONNECTION_FAIL_COUNT = "ReplicateDataCreateConnectionFailCount"
   val REPLICATE_DATA_CONNECTION_EXCEPTION_COUNT = "ReplicateDataConnectionExceptionCount"
+  val REPLICATE_DATA_FAIL_NON_CRITICAL_CAUSE_COUNT = "ReplicateDataFailNonCriticalCauseCount"
```
Comment (SteNicholas) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:148]: Update the dashboard json for `ReplicateDataFailNonCriticalCauseCount"`.

Comment (lyy-pineapple) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:148]: Thank you for creating an additional PR to update the JSON. I have limited understanding in this area.

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 2]
```diff
@@ -1752,7 +1752,7 @@ private StatusCode getPushDataFailCause(String message) {
     StatusCode cause;
     if (message == null) {
```
Comment (waitinfuture) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1753]: Shall we also check `PUSH_DATA_FAIL_NON_CRITICAL_CAUSE_REPLICA` here?

Comment (lyy-pineapple) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1753]: Thanks~, fixed

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.5.0)

**Pull request closed without merging**